### PR TITLE
[IZPACK-1700] Substance LaF for JDK >=9. Update jgoodies.

### PR DESCRIPTION
**Pull request by tia-ru**
Use "Radiance" for JDK>=9.  "Radiance" is evolution of "Substance"
https://github.com/kirill-grouchnikov/radiance

### CODE REVIEW

File: izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java
Comment (reinhapa) [inline izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: @akuhtz  do we not have a similiar implementation already?

Comment (tia-ru) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: I didn't found.
As soon as it's a copy from JDK it guaranteed to support all java version string. I suggest to use the JavaVersion class in
CompareVersionsCondition class too.
There is also problems with <javaversion> handler. It should supports modern jdk version schema (https://openjdk.org/jeps/223) so 1.8 and 8 should means the same version and be equals. JavaVersion#parseToJdk9Schema method may helps to implement such behaviour.
Taking this into account i commit full java version string parser, although my code use just major version part.

Comment (akuhtz) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: There is `SystemUtils` from commons-lang3 (https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/SystemUtils.html) that should allow to check whether running on Java 8 or newer.

Comment (tia-ru) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: Good point. The `org.apache.commons.lang3.JavaVersion` could solve it. Unfortunately JavaVersion#getJavaVersion(String) has package private access. We have to parse version string from xml config to JavaVersion some how.

Comment (akuhtz) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: @tia-ru If I read it correct, the following should detect if the current java verion is java 8 or less:
```
boolean isJava8OrLess = SystemUtils.isJavaVersionAtMost(JavaVersion.JAVA_1_8);
```

Comment (reinhapa) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: @tia-ru I would be in favor of destill the JavaVersion class down to just provide an `public static getMajorVersion()` Method. Within we could try to get access to the static method `Runtime.version()` Method and call reflectivly to it's `major()` method. If that fails we would then fall back to the old `String` based logic.

Comment (tia-ru) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: Ok. Then I switch to simple String based logic. Check next PR.
P.S. commons-lang3 is not deployed within installer. It has "test" maven scope.

File: izpack-installer/src/main/java/com/izforge/izpack/installer/container/provider/GUIInstallDataProvider.java [hunk 1]
```diff
@@ -375,16 +370,13 @@ private void loadGUIInstallData(GUIInstallData installData, Resources resources)
...
-            return versionString.charAt(2) - '0';
-        } else {
-            String[] split = versionString.split("[.+_-]", 2);
-            return Integer.parseInt(split[0]);
+        if (v.startsWith("1.")) {
+            v = v.substring(2);
         }
+        String[] split = v.split("[.+_-]", 2);
+        return Integer.parseInt(split[0]);
     }
```
Comment (reinhapa) [inline izpack-installer/src/main/java/com/izforge/izpack/installer/container/provider/GUIInstallDataProvider.java]: You could call the method `getCurrentMajorVersion()` from `izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java`  outlined above...

File: izpack-compiler/src/main/java/com/izforge/izpack/compiler/CompilerConfig.java [hunk 2]
```diff
@@ -4063,6 +4120,10 @@ private void logCombineOsLists(List<OsModel> parentOsList, List<OsModel> osList,
     private static int getJavaMajorVersion(String versionString) {
         if (versionString == null || versionString.isEmpty())
             return 1;
-        return JavaVersion.parseToJdk9Schema(versionString).feature();
+        if (versionString.startsWith("1.")) {
+            versionString = versionString.substring(2);
+        }
+        String[] split = versionString.split("[.+_-]", 2);
+        return Integer.parseInt(split[0]);
     }
```
[the remark this answers is not in the document]
Comment (tia-ru) [reply izpack-compiler/src/main/java/com/izforge/izpack/compiler/CompilerConfig.java]: ок

File: izpack-installer/src/main/java/com/izforge/izpack/installer/container/provider/GUIInstallDataProvider.java (hunk 1 above)
Comment (tia-ru) [reply izpack-installer/src/main/java/com/izforge/izpack/installer/container/provider/GUIInstallDataProvider.java]: Done. I mimic API of `java.lang.Runtime.Version`

File: izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java
Comment (reinhapa) [inline izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: @tia-ru could you finally add the copyright header as from compiler config and also a test for the new class would be nice...

**Review by reinhapa — CHANGES_REQUESTED** [review]

File: izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java
Comment (akuhtz) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: @reinhapa @tia-ru  There is something in `com.izforge.izpack.installer.requirement.JavaVersionChecker` that uses `CompareVersionsMajorCondition`. Maybe you can check it.

Comment (tia-ru) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: Done

Comment (tia-ru) [reply izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java:1]: This is a different task. Maybe later in another PR.

**Review by reinhapa — APPROVED** [review]

**Review by akuhtz — APPROVED** [review]

**Pull request merged**